xcube/webapi/ows/stac/controllers.py

```diff
@@ -744,17 +795,20 @@ def _get_dataset_feature(
     collection_id: str,
     feature_id: str,
     full: bool = False,
+    granted_scopes: Optional[set[str]] = None,
 ) -> dict:
-    bbox = GridBbox(ctx.get_ml_dataset(dataset_id).grid_mapping)
+    dataset_dict = get_dataset(
+        ctx, dataset_id, base_url=base_url, granted_scopes=granted_scopes
+    )
 
     return {
         "stac_version": STAC_VERSION,
         "stac_extensions": STAC_EXTENSIONS,
         "type": "Feature",
-        "id": feature_id,
-        "bbox": bbox.as_bbox(),
-        "geometry": bbox.as_geometry(),
-        "properties": _get_cube_properties(ctx, dataset_id),
+        "id": dataset_dict.pop("id"),
+        "bbox": dataset_dict.pop("bbox"),
+        "geometry": dataset_dict.pop("geometry"),
```

Comment:
_Popping_ may be dangerous here, because you cannot know whether a cached dictionary is returned from `get_dataset()` and you would change it in place then. Why don't you just _get_ those properties?

Reply:
I wanted to reduce duplication of information within the STAC item. But I see your point. So I changed `pop` to `get`.